lib/src/main/java/uk/gov/di/ipv/core/library/helpers/LogHelper.java

```diff
@@ -88,6 +92,22 @@ public static void logInfoMessageWithFieldAndValue(
         LoggingUtils.removeKey(logField.getFieldName());
     }
 
+    public static void logMessageWithFieldsAndValues(
+            Level logLevel,
+            String message,
+            LogField logField1,
+            String logFieldValue1,
+            LogField logField2,
+            String logFieldValue2,
+            LogField logField3,
+            String logFieldValue3) {
+        LoggingUtils.appendKey(logField1.getFieldName(), logFieldValue1);
+        LoggingUtils.appendKey(logField2.getFieldName(), logFieldValue2);
+        LoggingUtils.appendKey(logField3.getFieldName(), logFieldValue3);
+        LOGGER.log(logLevel, message);
+        LoggingUtils.removeKeys(logField1.getFieldName(), logField2.getFieldName(), logFieldValue3);
+    }
+
```

Comment:
I think we should be using a structured message rather than this mechanism.

Reply:
Yeah, i thought that. But the structured message blows up if you give it a value of `null`. The place this is being used is triggered by two scenarios and it means the response may not always have all the fields. I could add a bunch of null checks, but it might leave some slightly weird values in the logs. I think this way null values just don't get included.